Source/inv_iterators.hpp

```diff
@@ -19,7 +19,7 @@ class ItemsContainerRange {
 	class Iterator {
 	public:
 		using iterator_category = std::forward_iterator_tag;
-		using difference_type = void;
+		using difference_type = int;

```

Comment:
Can you elaborate why this is being changed? Was it wrong? Is it needed for the new iterator for some reason?

Reply:
`std::count_if` needs (and returns) `difference_type`